File: daily_problems/2025/06/0616/personal_submission/cf254c_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/all"

using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

// Let's set a bit and flow!
// I came, I divided, I conquered!
// using namespace __gnu_cxx;
// using namespace __gnu_pbds;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    ifstream fin("input.txt");
    ofstream fout("output.txt");

    string s1, s2;
    fin >> s1 >> s2;

    vector<int> cnt1(26, 0), cnt2(26, 0);
    for (auto &c: s1) cnt1[c - 'A'] ++;
    for (auto &c: s2) cnt2[c - 'A'] ++;

    int n = s1.size(), ans = 0;
    vector<int> v1(26, 0), v2(26, 0);
    for (int i = 0; i < 26; i ++) {
        int v = abs(cnt1[i] - cnt2[i]);
        if (cnt1[i] > cnt2[i]) v1[i] = v;
        else v2[i] = v;
        ans += v;
    }

    for (int i = 0; i < n; i ++) {
        cnt1[s1[i] - 'A'] --;

        if (v1[s1[i] - 'A']) {
            for (int j = 0; j < 26; j ++) {
                if (j >= s1[i] - 'A' && cnt1[s1[i] - 'A'] >= v1[s1[i] - 'A']) break;
                if (v2[j]) {
                    v1[s1[i] - 'A'] --;
                    v2[j] --;
                    s1[i] = 'A' + j;
                    break;
                }
            }
        }
    }

    fout << ans / 2 << '\n';
    fout << s1;

    return 0;
}

File: daily_problems/2025/06/0616/personal_submission/cf254c_yawn_sean.py
# Submission link: https://codeforces.com/contest/254/submission/324578291
def main():
    fin = open('input.txt', 'r')
    input = lambda: fin.readline().strip()

    fout = open('output.txt', 'w')
    print = lambda x: fout.write(x + '\n')

    s1 = [ord(c) - ord('A') for c in I()]
    s2 = [ord(c) - ord('A') for c in I()]

    cnt1 = [0] * 26
    for c in s1:
        cnt1[c] += 1

    cnt2 = [0] * 26
    for c in s2:
        cnt2[c] += 1

    n = len(s1)
    ans = 0

    v1 = [0] * 26
    v2 = [0] * 26

    for i in range(26):
        v = abs(cnt1[i] - cnt2[i])
        if cnt1[i] > cnt2[i]: v1[i] = v
        else: v2[i] = v
        ans += v

    for i in range(n):
        cnt1[s1[i]] -= 1
        
        if v1[s1[i]]:
            for j in range(26):
                if j >= s1[i] and cnt1[s1[i]] >= v1[s1[i]]: break
                if v2[j]:
                    v1[s1[i]] -= 1
                    v2[j] -= 1
                    s1[i] = j
                    break

    print(str(ans // 2))
    print(''.join(chr(ord('A') + c) for c in s1))